tuplex/codegen/include/BlockGeneratorVisitor.h

```diff
@@ -96,7 +97,7 @@ namespace codegen {
 
             // simplify interfaces a bit
             inline codegen::SerializableValue load(llvm::IRBuilder<>& builder) const {
-                assert(ptr && sizePtr);
+                // assert(ptr && sizePtr);
```

Comment:
can we remove this? or why is it commented?

Reply:
maybe keep this assertion if it's not an iterator, otherwise check that it's an iterator? i.e. update the assertion rather than removing it altogether

Reply:
Kept the assertion.
Changed to use dummy LLVM type for iterators during declareVariables() so iterator's slot ptr won't become nullptr.